---
title: "BIO782P - Week 1 Assignment"
author: "Ana Penedos (UID 1932)"
date: "14th December 2018"
output:
  html_notebook: default
   # latex_engine: xelatex
    number_sections: no
  html_document:
    toc: no
  pdf_document:
    fig_caption: no
    fig_height: 4
    fig_width: 5
    keep_tex: no
#monofont: Monaco, FreeMono
monofont: FreeMono
#mainfont: Helvetica Light, Helvetica
mainfont: Helvetica
subparagraph: yes
geometry: a4paper, margin=20mm
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# packages used:
install.packages("ggplot2")
install.packages("plyr")
install.packages("car")
```

# Dataset 1: Marine microbial diversity

* 2 cruises: Jan and Aug
* sampled 20l of seawater
* 10 separate locations 
* equatorial (0-8 degrees N) and temperate (48-55 degrees N) waters
* samples were filtered to enrich the microbiome
* DNA extracted, WGS sequenced, and sequence reads identified using BLASTN
* Microbial diversity assessed using UniFrac, expressed relative to reference

## Data import
The data was imported into Microsoft Excel and checked for obvious issues such as missing values. As none were found, we can import it into R.
```{r}
marine <- read.table("../../labs/Assessment_1/Student_data/2018/part_1_student_1932.tdf", 
                     sep = "\t",
                     header = TRUE)
head(marine) # check if there are obvious issues with data import
names(marine) # get the names of data frame variables
str(marine) # check number of observations, variables and their type
```

Season and latitude are factors with two levels each, matching the described experiment.

```{r}
# Defining variables to simplify dataframe references:
diversity.col <-marine$UniFracInd
latitude.col <- marine$latitude
season.col <- marine$season
```

## 1. How does microbial diversity change with latitude?
```{r}
library(ggplot2)
# set up plot-specific elements
# ggplot object plotting UniFracInd against latitude
marine.latitude.plot <- ggplot(marine,
                             aes(factor(latitude.col), 
                                 diversity.col,
                                 fill = latitude.col))
# specific plot title and axes labels
marine.latitude.titles <- labs(title = "Microbial Density vs. Latitude",
                               x = "Latitude",
                               y = "UniFracInd")
# change fill from the default colours
latitude.fill <- scale_fill_manual(values=c("#009999", "#0099FF"))

# set up re-usable elements to edit violin plots
# create a violin plot with count on the y axis and including outliers
violin.plot <- geom_violin(scale = "count",
                           trim = FALSE)
## add sample points
#geom_jitter(height = 0, width = 0.2) +
# add a boxplot
add.boxplot <- geom_boxplot(width = 0.1,
                            fill = "white")
# add the mean of values
add.mean <- stat_summary(fun.y = mean,
                         geom = "point",
                         shape = 23,
                         size = 2,
                         fill = "red")
# change the plot theme to classic (no grey plot area or gridlines)
set.plot.theme <- theme_classic()
# set variable for the style of plot titles;
# hjust is the horizontal position of the title
title.style <- element_text(face = "bold.italic",
                            size = 12,
                            hjust = .5)
# set variable for the style of plot axes
axes.labels.style <- element_text(face = "bold",
                                  size = 10)
# apply title and axes format to the plot theme and remove legend
title.axes.theme <- theme(plot.title = title.style,
                          axis.title = axes.labels.style,
                          legend.position="none")

# bring plot and formatting together
marine.latitude.plot + marine.latitude.titles + violin.plot + add.boxplot + 
  add.mean + latitude.fill + set.plot.theme + title.axes.theme
```
The violin plot shows that although the median of UniFracInd is identical between samples taken in equatorial waters and those obtained in tropical waters, the distribution of sample microbiological diversity is slightly different in each type of environment. While in temperate waters, the samples with a UniFracInd above the median tend to aggregate close to the median value, with the remaining values more spread below the mid point's value, in tropical waters this trend is reversed, with the samples with higher diversity than the median value more spread and those with lower than median diversity showing UniFracInd values closer to the median value. This is illustrated by the fact that the means of each type of sample (red losanges) difer by a larger margin than the medians, with samples collected in temperate waters showing a lower mean of microbial diversity index (UniFracInd) than those from tropical waters.
In order to test whether these differences are significant we could perform a t-test. The null hypothesis would be:
H0: There is no difference in mircobial diversity between samples collected in temperate and tropical regions.
The alternative hypothesis would be:
Ha: There is a difference in mircobial diversity between samples collected in temperate and tropical regions.
Significance level, a=0.05
t-test assumptions:
1. The data is measured in a continuous scale - UniFracInd is a continuous variable;
2. The samples collected are random and representative - 20l of water from different locations;
3. The data is normally distributed;
4. Large size sample;
5. Homogeneous variance.

From the histograms below, the total data is close to the normal distribution:
```{r}
hist(diversity.col)
```
However, the sub-sets of data corresponding to the temperate and tropical samples appear farther from the normal distribution (assumption 3).
```{r}
par(mfrow=c(1,2))
hist(diversity.col[latitude.col=="temperate"])
hist(diversity.col[latitude.col=="tropical"])
```
We can perform a normality test on these:
```{r}
shapiro.test(diversity.col[latitude.col=="temperate"])
shapiro.test(diversity.col[latitude.col=="tropical"])
```
The p-value is above the recommended for this test (<0.1; [ethz R manual](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/shapiro.test.html), which means we cannot exclude the null hypothesis for the Shapiro-Wilk test that the sample is normally distributed, so we can proceed to assess whether the remaining assumptions of the t-test are met.

We can determine the number of samples in each group:
```{r}
library(plyr)
latitude.num.samples <- count(marine, "latitude")
latitude.num.samples

```
There are 20 samples in each group, which should be sufficient if the other assumptions are met.

Homoscedasticity of variance can be tested with Levene's test, which is a variance homogeneity test that can be used when there is a slight departure from normality [R cookbook](http://www.cookbook-r.com/Statistical_analysis/Homogeneity_of_variance/):
```{r}
library(car)
leveneTest(diversity.col ~ latitude.col)
```
Given that the null hypothesis for the Levene's homoscedasticity test is that the variances are homogeneous, the p-value of 0.391 is not sufficient to reject homoscedasticity.

Given that we can assume all t-tests assumptions are met, we can then perform a t-test as below:
```{r}
t.test(diversity.col[latitude.col=="temperate"], diversity.col[latitude.col=="tropical"])
```

The p-value of 0.1209 would indicate that the likelihood of observing the current samples if H0 is true is above the set significance level of 0.05 and hence we cannot reject the null hypothesis. From these analyses, I would conclude there is no statistically significant difference in microbial diversity at different latitudes.

## 2. How does microbial diversity change with time of year?
```{r}
# set up plot-specific elements
# ggplot object plotting UniFracInd against season
marine.season.plot <- ggplot(marine,
                             aes(factor(season.col), 
                                 diversity.col,
                                 fill = season.col))
# specific plot title and axes labels
marine.season.titles <- labs(title = "Microbial Density vs. Season",
                             x = "Season",
                             y = "UniFracInd")
# change fill from the default colours
season.fill <- scale_fill_manual(values=c("#FF9900", "#6699FF"))

# bring plot and formatting together
marine.season.plot + marine.season.titles + violin.plot + add.boxplot + 
  add.mean + season.fill + set.plot.theme + title.axes.theme
```
Higher microbial diversity is observed in the August cruise than in January. The values for UniFracInd are more normally distributed around the median value than those measured by latitude. As such, median and mean values coincide, being higher in August. The spread of UniFracInd values measured in January is smaller than that seen in August.

We can perform a t-test to assess if these differences are significant.
H0: There is no difference in mircobial diversity between samples collected in Aug and Jan seasons.
The alternative hypothesis would be:
Ha: There is a difference in mircobial diversity between samples collected in Aug and Jan seasons.
Significance level, a=0.05

As for the latitude, we can assume that conditions 1 and 2 for a t-test are verified.
Normality:
```{r}
par(mfrow=c(1,2))
hist(diversity.col[season.col=="Aug"])
hist(diversity.col[season.col=="Jan"])
```
```{r}
shapiro.test(diversity.col[season.col=="Aug"])
shapiro.test(diversity.col[season.col=="Jan"])
```
These two groups of observations appear to be normally distributed (p>0.1).

Large size sample:
```{r}
season.num.samples <- count(marine, "season")
season.num.samples
```
Again, there are are 20 samples in each group, which should be sufficient for a t-test if the remaining assumptions are met.

Homoscedasticity:
```{r}
leveneTest(diversity.col ~ season.col)
```
The probability of observing the current variances in the two groups if they are homoscedastic is very high (>0.7) by Levene's test, hence we can assume there is homogeneity in the variances.

We can thus perform a t-test:
```{r}
t.test(diversity.col[season.col=="Aug"], diversity.col[season.col=="Jan"])
```

This time, we obtain a p-value < 0.02, below the sgnificance level set (a=0.05). In this context, we would reject the null hypothesis that there is no difference in microbial diversity in each season, and accept the alternative hypothesis that states that there is a statistically significant difference between microbial diversity in Jan and Aug. From the violin plot, we can see that there is higher diversity in Aug than in Jan.

## 3. Is there an interaction between the season, and location?
To statistically assess whether there is an interaction of season and latitude, we can fit a linear model which takes into account interactions of season and latitude and then do an analysis of variance to determine which factors better explain changes in diversity.

```{r}
# Correlation of diversity with an interaction of latitude and season
corr.seas.lat <- lm(diversity.col ~ season.col * latitude.col)
anova(corr.seas.lat)
```
From the ANOVA test of the model fitted taking into acount the interaction of latitude and season on the measure microbial diversity we conclude that, as was apparent from the simpler t-tests, season has a larger impact on microbial diversity than latitude. The term acounting for interaction between season and latitude has a p-valua > 0.1, which has no significance level. This suggests that the main impact on microbial diversity comes from the season when the samples collected and that we cannot exclude the hypothesis that there is no significant interaction between season and location.

# Dataset 2: Pairwise nucleotide substitutions and RNA expression levels

* luciferase
* RNA-seq project
* over 900 coding loci from 208 species of plants seq'd
* found 30 genes with homology to luciferase
* measured exprssion levels in each species
* phylogenetic tree
* computed pairwise genetic distances each species' gene and that of closely-related *Arabidopsis thaliana*

## Data import
No issues were observed in Excel.
```{r}
luciferase <- read.table("../../labs/Assessment_1/Student_data/2018/part_2_student_1932.tdf",
                         sep = "\t",
                         header = TRUE)
head(luciferase) # quick check of correct data import
names(luciferase) # check names of columns for dataset
str(luciferase) # check that factors and variables were assigned correctly
```
```{r}
# simplify dataframe references
luciferase.expression <- luciferase$expression_fold
aa.substitutions <- luciferase$distance
```
## 1. How does the putative 'luciferase' homologue expression change with genetic distance (amino acid substitutions)?
We can start by visualising these data using a scatter plot:
```{r}
plot(aa.substitutions, luciferase.expression)
```
There appears to be a low correlation of luciferase expression with amino acid substitutions. We can fit a model and analyse the residuals plots.

```{r}
expr.dist.model <- lm(luciferase.expression ~ aa.substitutions)
summary(expr.dist.model)
```
From the summary of the model
```{r}
par(mfrow=c(2,2))
plot(expr.dist.model)
```


## 2. Comment on whether the model assumptions are valid.


## 3. Assuming your model is *statistically* valid, can you guess what effect is responsible for the relationship you've found?

## Dataset 3: HIV viral load and within-patient population dynamics

Populations of the HIV virus are able to evade drug therapy and persist in certain immunoprotected tissues in the body, principally areas of the central nervous system (CNS) where CD4+ T-cells are prevented from entering. An HIV+ patient has enrolled in a study and consented to viral load sampling over 40 weeks in a year. Each week a sample was taken from the brain or spinal cord, and the viral population size ('load', here expressed as log10(number of viral particles per ml)) measured using a chip assay.  Average Shannon population diversity, and mean pairwise genetic distance from individual viruses present in each weekly sample to a reference sequence was assessed using single-copy PCR amplification of the viral Env gene, and we hope to find some relationship between population size, diversity, evolutionary distance and tissue.

The sampling (involing an epidural needle inserted deep into the CNS) is painful and carries a high risk of complications, so we need to make sure we make the most of this data. **Do your best to fit a model explaining viral load in terms of the other variables, using any combination of variables you see fit, and any model selection procedure. Write up your results accordingly, with any figures which are appropriate for the research paper we plan to write.** Since we're unsure how likely the host immune system is to penetrate the CNS, CD4+ cell counts in the patient's general circulation were assessed using flow cytometry, and categorized as 'low' or 'high' - you may wish to consider these in your analysis. The rows in the data table are in chronological order.

## General feedback from previous years.

Overall, we have been impressed with the standard of previous reports. Most of you manage to get the analyses more or less right, and most of you produce good- quality scripts. None of them had too much annotation - remember that when you’re annotating a script you’re really writing a guide to yourself describing what it does. Think about the information that might be useful if you come back to the analysis a year or two in the future.

Regarding the "results section" of the reports, these are a lot more variable. Many of you put too much analysis into these, and in particular include material that would not normally go in a journal results section. Preliminary and exploratory analysis, diagnostic plots and the like are not generally included in journal results sections. Many of you have redundant graphs – you shouldn’t show the reader the same set of data twice. A common problem is too much focus on the statistics and not enough on what the statistical results mean in terms of the biology of the system, or in terms of effect sizes: don’t just tell me that there’s an effect, tell me how big the effect is, and put it in meaningful biological terms.

Figure captions are something that most of you need to work on. When you’re writing a figure caption, it’s a good idea to try to write it so that a casual reader who is skimming through the paper can look at the figure, have a look at the caption and have at least a rough idea about what the figure is showing. That doesn’t mean that each figure should have the whole methods section reproduced, but you can include a sentence or two that gives the casual reader a basic idea of what’s going on.

Other common problems include:

 * Including p-values without test statistics or degrees of freedom
 * Including multiple tests of the same thing (e.g. using a post-hoc test on a fitted
model, collapsing two factor levels and then comparing models with a partial F-
test)
 * Carrying out tests for normality on data prior to analysis (skipping data exploration/eyeballing and )
 * Mixing p-values and AIC as criteria for model selection - the philosophy behind
these is very different and you should use one or the other but not both.
 * Giving significance levels for main effects when they are also included in higher-
order interaction terms
 * Including graphs showing no effect. There are circumstances when you might want
to do this, if you’re reporting a negative result and you think it’s necessary to make a point about how little relationship there is, but in general we wouldn’t put this sort of thing in.
 * Including code, function names etc. from R. Results sections from journals wouldn’t usually include this kind of material unless you’re describing an esoteric analysis that the readers will not be familiar with, in which case you might say “We fitted a generalised additive model to the data using the gam() function as implemented in the mcgv package (Wood 2014)” but usually you would just tell the reader the type of analysis used.
 * Referring to “non-significant” results as “insignificant”. Don’t do this - have a think about why.
 * No references! Not a single person put a reference in their results section.

\pagebreak 
